index.js

```diff
@@ -2441,6 +2441,30 @@
         return n < 0 || negativeZero(n) ? Nothing() : slice(0, -n, xs);
       });
 
+  //# reverse :: [a] -> [a]
+  //.
+  //. Returns the elements of the given list in reverse order.
+  //.
+  //. ```javascript
+  //. > S.reverse([1, 2, 3])
+  //. [3, 2, 1]
+  //. > S.reverse('abc')
+  //. 'cba'
+  //. ```
+  S.reverse =
+  def('reverse',
+      {},
+      [List(a), List(a)],
+      function reverse(xs) {
+          var result = [];
+          if (xs.length === 0) return xs;
```

Comment:
I'd prefer not to include this optimization. It's nice to return a new array for all array inputs.